### ISSUE TITLE
[MiniLLM] mismatch between formula and implementation (gradL)_long? [old_logprobs not find in paper]

### ISSUE THREAD

**Issue by lancerts** [microsoft/LMOps#265]
In paper
<img width="600" alt="Screenshot 2024-09-10 at 4 22 08 PM" src="https://github.com/user-attachments/assets/88c7d527-d9e8-44b3-a4b2-8f9ba70447b9">,
in code
```
  def _pg_loss(
      self,
      logprobs: TensorType["batch_size", "response_size"],
      old_logprobs: TensorType["batch_size", "response_size"],
      advantages: TensorType["batch_size", "response_size"],
      mask: TensorType["batch_size", "response_size"],
      w: TensorType["batch_size", "response_size"],
  ):
      """PPO objective function.
      References:
      - https://stable-baselines.readthedocs.io/en/master/modules/ppo2.html
      """
      n = mask.sum()
      
      log_ratio = (logprobs - old_logprobs) * mask
      ratio = torch.exp(log_ratio.float())            
      ratio = ratio * w
```
Is `old_logprobs` referring to the teacher-mixed sampling $\tilde{p}$ or its something else?

**Comment by t1101675** [comment]
`old_logprobs` refers to $q_{\theta}$, which will not be taken derivatives. `logprobs` is $q_{\theta}$ that will be taken derivatives. Computing gradients for `ratio` refers to compute $\frac{\nabla q_{\theta}}{q_{\theta}}$. Therefore, `ratio * w` refers to $\rho_t(\theta) = \frac{q_{\theta}}{\widetilde{p}}$, where only the $\theta$ in the numerator will be taken gradients.

**Comment by lancerts** [comment]
Thanks for the detailed explanation.
